src/Installers/Windows/WindowsHostingBundle/Bundle.wxs

```diff
@@ -57,13 +57,35 @@
         <!-- Bundle variables -->
         <Variable Name="BundleNameShort" Value="$(var.BundleNameShort)"/>
         <Variable Name="BundleNameSub" Value="$(var.BundleNameSub)"/>
-        <Variable Name="OPT_NO_ANCM" Value="0" bal:Overridable="yes"/>
-        <Variable Name="OPT_NO_FTS" Value="0" bal:Overridable="yes"/>
-        <Variable Name="OPT_NO_LTS" Value="0" bal:Overridable="yes"/>
-        <Variable Name="OPT_NO_SHAREDFX" Value="0" bal:Overridable="yes"/>
-        <Variable Name="OPT_NO_RUNTIME" Value="0" bal:Overridable="yes"/>
-        <Variable Name="OPT_NO_X86" Value="0" bal:Overridable="yes"/>
-        <Variable Name="OPT_NO_SHARED_CONFIG_CHECK" Value="0" bal:Overridable="yes" />
+        <Variable Name="OPT_NO_ANCM" bal:Overridable="yes"/>
+        <Variable Name="OPT_NO_FTS" bal:Overridable="yes"/>
+        <Variable Name="OPT_NO_LTS" bal:Overridable="yes"/>
+        <Variable Name="OPT_NO_SHAREDFX" bal:Overridable="yes"/>
+        <Variable Name="OPT_NO_RUNTIME" bal:Overridable="yes"/>
+        <Variable Name="OPT_NO_X86" bal:Overridable="yes"/>
+        <Variable Name="OPT_NO_SHARED_CONFIG_CHECK" bal:Overridable="yes" />
+
+        <!-- Use values from the registry for switches, if the registry values are set & the user hasn't passed any switches from the command line -->
+
+        <?define Options=OPT_NO_ANCM;OPT_NO_FTS;OPT_NO_LTS;OPT_NO_SHAREDFX;OPT_NO_RUNTIME;OPT_NO_X86;OPT_NO_SHARED_CONFIG_CHECK ?>
+        <?define RegistrySeachCondition="NOT (OPT_NO_ANCM OR OPT_NO_ANCM=0 OR OPT_NO_FTS OR OPT_NO_FTS=0 OR OPT_NO_LTS OR OPT_NO_LTS=0 OR OPT_NO_SHAREDFX OR OPT_NO_SHAREDFX=0 OR OPT_NO_RUNTIME OR OPT_NO_RUNTIME=0 OR OPT_NO_X86 OR OPT_NO_X86=0 OR OPT_NO_SHARED_CONFIG_CHECK OR OPT_NO_SHARED_CONFIG_CHECK=0)" ?>
+
+        <?foreach Option in $(var.Options) ?>
+            <util:RegistrySearch Id="$(var.Option)_should_be_set" 
+                             Condition="$(var.RegistrySeachCondition)"
+                             Root="HKLM" 
+                             Key="Software\Microsoft\dotnet\host\options\$(var.MajorVersion).$(var.MinorVersion)" 
+                             Value="$(var.Option)" 
+                             Result="exists" 
+                             Variable="$(var.Option)_Should_Be_Set"/>
```

Comment:
Can we rename this to '$(var.Option)_Exists' - that's the convention we typically used for existence checks. Sorry I didn't catch this earlier. It also make the second search condition a bit more readable since right now we have to similarly named variables that differ by "S" and "s"

Reply:
Sure